lib/experimental/block-editor-settings-mobile.php

```diff
@@ -31,6 +31,8 @@ function gutenberg_get_block_editor_settings_mobile( $settings ) {
 		$settings['__experimentalEnableQuoteBlockV2'] = true;
 		// To tell mobile that the site uses list v2 (inner blocks).
 		$settings['__experimentalEnableListBlockV2'] = true;
+		// To tell mobile if it should mark the Gallery block as Unsupported for older versions of WordPress.
+		$settings['__unstableGalleryWithImageBlocks'] = is_wp_version_compatible( '5.9' );
```

Comment:
To be honest, I'm not really sure what that's doing but let me ping @glendaviesnz do you know by any chance what `(int) get_option( 'use_balanceTags' ) !== 1` does? And if that would be needed for mobile? I think with just having `is_wp_version_compatible( '5.9' )` would be enough but it'd be nice to confirm, thank you!

Reply:
In versions of WP < 5.9 if `use_balanceTags` was enabled the post HTML was parsed and rewritten to make sure all the HTML tags were correctly balanced, but the code doing this didn't accept `figure` as being nestable so would rewrite the gallery code to remove the nesting. 

This was [updated here](https://github.com/WordPress/wordpress-develop/pull/1678) so that the gallery code was left nested as is even if `use_balanceTags` is enabled on a site. So, as long as a site is >= 5.9 there should be no issues even if `use_balanceTags` is enabled.